delfin/drivers/hds/vsp/rest_client.py

```diff
@@ -0,0 +1,318 @@
+    hdsvsp_Volumes_Url = '/ConfigurationManager/v1/objects/ldevs'
+    # GET
+    hdsvsp_Ports_Url = '/ConfigurationManager/v1/objects/ports'
+    hdsvsp_Logout_Url = '/ConfigurationManager/v1/objects/sessions/'
+    hdsvsp_Alert_Url = '/ConfigurationManager/v1/objects/sessions/'
+
+    hdsvsp_Auth_Key = 'Authorization:Session'  # X-HP3PAR-WSAPI-SessionKey
+
+    def __init__(self, **kwargs):
+
+        rest_access = kwargs.get('rest')
+        if rest_access is None:
+            raise exception.InvalidInput('Input rest_access is missing')
+        host = rest_access.get('host')
+        port = rest_access.get('port')
+        self.san_user = rest_access.get('username')
+        self.san_password = rest_access.get('password')
+        self.san_address = 'https://' + host + ':' + port
+        self.session = None
+        self.url = self.san_address
+        self.enable_verify = False
+        self.ca_path = 'certificate verification file path'
+
+    def init_http_head(self):
+        self.session = requests.Session()
+        self.session.headers.update({
+            "Connection": "keep-alive",
+            'Accept': 'application/json',
+            "Content-Type": "application/json"})
+        self.session.verify = False
+        if not self.enable_verify:
+            self.session.verify = False
+        else:
+            LOG.debug("Enable certificate verification, ca_path: {0}".format(
+                self.ca_path))
+            self.session.verify = self.ca_path
+            # if not self.assert_hostname:
+            #     self.session.mount("https://", HostNameIgnoreAdapter())
+        self.session.trust_env = False
+
+    def do_call(self, url, data, method,
+                calltimeout=consts.SOCKET_TIMEOUT):
+        """Send requests to hdsvsp storage server.
+        """
+        if 'http' not in url:
+            if self.url:
+                url = self.url + url
+
+        LOG.info('obj url=={0}'.format(url))
+        kwargs = {'timeout': calltimeout}
+        if data:
+            kwargs['data'] = json.dumps(data)
+
+        if method in ('POST', 'PUT', 'GET', 'DELETE'):
+            func = getattr(self.session, method.lower())
+        else:
+            msg = _("Request method %s is invalid.") % method
+            LOG.error(msg)
+            raise exception.StorageBackendException(reason=msg)
+        res = None
+        try:
+            res = func(url, **kwargs)
+        except requests.exceptions.ConnectTimeout as ct:
+            LOG.error('ConnectTimeout err: {}'.format(ct))
+            raise exception.ConnectTimeout(reason=ct)
+        except Exception as err:
+            LOG.exception('Bad response from server: %(url)s.'
+                          ' Error: %(err)s', {'url': url, 'err': err})
+            raise err
+
+        return res
+
+    def call(self, url, data=None, method=None):
+        """Send requests to server.
+        If fail, try another RestURL.
+        """
+        accessSession = None
```

Comment:
avoid name of variable in camel case style in python. Can be changed to access_session